dss/storage/validator.py

```diff
@@ -0,0 +1,78 @@
+"""Utilities in this file are used to removed extra fields from index data before adding to the index."""
+import re
+from jsonschema.validators import create, RefResolver
+from jsonschema import _utils, _validators
+
+import json
+from urllib.request import urlopen
+from urllib.error import URLError
+import ssl
+from typing import List
+import logging
+
+
+def read_json_url(link):
+    gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    f = urlopen(link, context=gcontext)
+    return json.load(f)
+
+
+def extract_json_fields(json_data: dict, path: List[str], fields: List[str]):
+    """
+        json_data: The json data from which to remove the field
+        path: The path through the json data to find the field.
+        fields: A list of fields names to remove from the json data at the locations specified by path."""
+    current = json_data
+    for next in path:
+        current = current[next]
+    for field in fields:
+        current.pop(field)
+
+DSS_Draft4Validator = create(
+    meta_schema=_utils.load_schema("draft4"),
+    validators={
+        u"$ref": _validators.ref,
+        u"additionalProperties": _validators.additionalProperties,
+        u"properties": _validators.properties_draft4,
+        u"required": _validators.required_draft4,
+    },
+    version="draft4",
+)
+
+resolver = RefResolver(referrer='', base_uri='')
+
+def scrub_index_data(index_data: dict, bundle_id: str, logger: logging.Logger) -> list:
+    extra_fields = []
+    extra_files = []
+    for file in index_data.keys():
+        core = index_data[file].get('core', None)
+        if core is not None:
+            try:
+                schema = read_json_url(core['schema_url'])
+            except URLError as ex:
+                extra_files.append(file)
+                logger.warning("%s", f"Unable to retrieve schema from url {core['schema_url']} due to exception: {ex}.")
+                continue
+
+            for error in DSS_Draft4Validator(schema, resolver=resolver).iter_errors(index_data[file]):
+                if error.validator == 'additionalProperties':
+                    path = error.path
+                    path.appendleft(file)
```

Comment:
It's better not to modify a data structure your being given primarily for informational purposes:

`path = [ *error.path, path ]` should work unless you absolutely need path to be a `deque`.

Reply:
I knew there was a better way of doing this. I forgot about using the `*` operator. I changed it to `path = [ document, *error.path]`